service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandler.java

```diff
@@ -62,6 +75,23 @@ public void sendNotificationToHmctsAdmin(SubmittedCaseEvent event) {
         sendNotification(HMCTS_COURT_SUBMISSION_TEMPLATE, email, parameters, reference);
     }
 
+    @EventListener
+    public void sendNotificationForC2Upload(final C2UploadedEvent caseEvent) {
+        List<String> roles = userDetailsService.getUserDetails(caseEvent.getAuthorization()).getRoles();
+
+        if (!roles.containsAll(UserRole.HMCTS_ADMIN.getRoles())) {
+            CaseDetails caseDetailsFromEvent = caseEvent.getCallbackRequest().getCaseDetails();
```

Comment:
Consider using ObjectMapper to map this to CaseData. That way we do not need to force the cast on line 84.

`String localAuthorityCode = caseData.getCaseLocalAuthority()`